AmplifyPlugins/Auth/AWSCognitoAuthPluginTests/AuthenticationProviderTests/AuthenticationProviderSigninWithWebUITests.swift

```diff
@@ -6,11 +6,229 @@
 //
 
 import Foundation
+import SafariServices
 
 import XCTest
 @testable import Amplify
 @testable import AWSCognitoAuthPlugin
 @testable import AWSMobileClient
 
+// swiftlint:disable:next type_name
 class AuthenticationProviderSigninWithWebUITests: BaseAuthenticationProviderTest {
+
+    var window: UIWindow {
+        let window = UIWindow()
+        window.rootViewController = MockRootUIViewController()
+        return window
+    }
+
+    /// Test a signIn with valid inputs
+    ///
+    /// - Given: Given an auth plugin with mocked service.
+    ///
+    /// - When:
+    ///    - I invoke signIn with valid values
+    /// - Then:
+    ///    - I should get a .done response
+    ///
+    func testSuccessfulSignIn() {
+
+        let mockSigninResult = UserState.signedIn
+        mockAWSMobileClient?.showSignInMockResult = .success(mockSigninResult)
+        let options = AuthWebUISignInRequest.Options()
+
+        let resultExpectation = expectation(description: "Should receive a result")
+        _ = plugin.signInWithWebUI(presentationAnchor: window, options: options) { result in
+            defer {
+                resultExpectation.fulfill()
+            }
+            switch result {
+            case .success(let signinResult):
+                guard case .done = signinResult.nextStep else {
+                    XCTFail("Result should be .done for next step")
+                    return
+                }
+                XCTAssertTrue(signinResult.isSignedIn, "Signin result should be complete")
+            case .failure(let error):
+                XCTFail("Received failure with error \(error)")
+            }
+        }
+        wait(for: [resultExpectation], timeout: apiTimeout)
+    }
+
+    /// Test a signIn with valid inputs
+    ///
+    /// - Given: Given an auth plugin with mocked service.
+    ///
+    /// - When:
+    ///    - I invoke signIn with valid values
+    ///    - Mock service returns invalid response
```

Comment:
should be invalid

Reply:
Not sure what you meant, anyways I updated the comment to mention that service return an invalid response like signedOut.